apiserver/config.go

```diff
@@ -0,0 +1,41 @@
+package apiserver
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/glog"
+
+	"istio.io/manager/model"
+)
+
+// Config is the complete configuration including a parsed spec
+type Config struct {
+	// Type SHOULD be one of the kinds in model.IstioConfig; a route-rule, ingress-rule, or destination-policy
+	Type string      `json:"type,omitempty"`
+	Name string      `json:"name,omitempty"`
+	Spec interface{} `json:"spec,omitempty"`
+	// ParsedSpec will be one of the messages in model.IstioConfig: for example an
+	// istio.proxy.v1alpha.config.RouteRule or DestinationPolicy
+	ParsedSpec proto.Message `json:"-"`
+}
+
+func (c *Config) parseSpec() error {
```

Comment:
Make it public and use in `istioctl`. We should parse proto on the client-side, and skip sending it to the API server.